tests/stream_test.ts

```diff
@@ -0,0 +1,810 @@
+    assert(a);
+    addedIds.push(a);
+    msgCount++;
+
+    const consumer = "someconsumer";
+    const xid = ">";
+    const data = await client.xreadgroup(
+      [{ key, xid }],
+      { group, consumer },
+    );
+
+    // each group should see ALL the messages
+    // that have been emitted
+    const toCheck = data[0].messages;
+    assertEquals(toCheck.length, msgCount);
+  }
+
+  for (const g of groups) {
+    assertEquals(await client.xgroup_destroy(key, g), 1);
+  }
+
+  await cleanupStream(client, key);
+});
+
+test("xrange and xrevrange", async () => {
+  const key = randomStream();
+  const firstId = await client.xadd(key, "*", { "f": "v0" });
+  const basicResult = await client.xrange(key, "-", "+");
+  assertEquals(basicResult.length, 1);
+  assertEquals(basicResult[0].xid, firstId);
+  assertEquals(basicResult[0].field_values.get("f"), "v0");
+
+  const secondId = await client.xadd(key, "*", { "f": "v1" });
+  const revResult = await client.xrevrange(key, "+", "-");
+
+  assertEquals(revResult.length, 2);
+  assertEquals(revResult[0].xid, secondId);
+  assertEquals(revResult[0].field_values.get("f"), "v1");
+  assertEquals(revResult[1].xid, firstId);
+  assertEquals(revResult[1].field_values.get("f"), "v0");
+
+  // count should limit results
+  const lim = await client.xrange(key, "-", "+", 1);
+  assertEquals(lim.length, 1);
+  const revLim = await client.xrevrange(key, "+", "-", 1);
+  assertEquals(revLim.length, 1);
+
+  await cleanupStream(client, key);
+});
+
+test("xclaim and xpending, all options", async () => {
+  await withConsumerGroup(async (key, group) => {
+    // xclaim test basic idea:
+    // 1. add messages to a group
+    // 2. then xreadgroup needs to define a consumer and read pending
+    //    messages without acking them
+    // 3. then we need to sleep 5ms and call xpending
+    // 4. from here we should be able to claim message
+    //    past the idle time and read them from a different consumer
+
+    await Promise.all(
+      [
+        client.xadd(key, 1000, { "field": "foo" }),
+        client.xadd(key, 2000, { "field": "bar" }),
+      ],
+    );
+
+    const initialConsumer = "someone";
+
+    const firstReply = await client.xreadgroup(
+      [{ key, xid: ">" }],
+      { group, consumer: initialConsumer },
+    );
+
+    const firstPending = await client.xpending(key, group);
+
+    await sleepMillis(5);
+
+    assertEquals(firstPending.count, 2);
+    assertNotEquals(firstPending.startId, firstPending.endId);
+    assertEquals(firstPending.consumers.length, 1);
+    assertEquals(firstPending.consumers[0].name, "someone");
+    assertEquals(firstPending.consumers[0].pending, 2);
+
+    const minIdleTime = 4;
+
+    // minimum options
+    const claimingConsumer = "responsible-process";
+    const firstClaimed = await client.xclaim(
+      key,
+      { group, consumer: claimingConsumer, minIdleTime },
+      1000,
+      2000,
+    );
+    switch (firstClaimed.kind) {
```

Comment:
Can this switch statement be replaced with the following code?
```suggestion
    assert(firstClaimed.kind === "messages");
    assertEquals(firstClaimed.messages.length, 2);
    assertEquals(
      firstClaimed.messages[0].field_values,
      new Map(Object.entries({ "field": "foo" })),
    );
...
```